app/components/CreateCrashRecordModal.tsx

```diff
@@ -0,0 +1,372 @@
+import Button from "react-bootstrap/Button";
+import Card from "react-bootstrap/Card";
+import Form from "react-bootstrap/Form";
+import Modal from "react-bootstrap/Modal";
+import Spinner from "react-bootstrap/Spinner";
+import AlignedLabel from "@/components/AlignedLabel";
+import FormControlDatePicker from "@/components/FormControlDatePicker";
+import { FaCirclePlus, FaCircleMinus } from "react-icons/fa6";
+import { useForm, SubmitHandler, useFieldArray } from "react-hook-form";
+import { useQuery, useMutation } from "@/utils/graphql";
+import { UNIT_TYPES_QUERY } from "@/queries/unit";
+import { CREATE_CRASH } from "@/queries/crash";
+import { LookupTableOption } from "@/types/relationships";
+import { Crash } from "@/types/crashes";
+import { useAuth0 } from "@auth0/auth0-react";
+
+interface CreateCrashModalProps {
+  /**
+   * A callback fired when either the modal backdrop is clicked, or the
+   * escape key is pressed
+   */
+  onClose: () => void;
+  /**
+   *  Callback fired after the user form is successfully submitted
+   */
+  onSubmitCallback: () => void;
+  /**
+   * If the modal should be visible or hidden
+   */
+  show: boolean;
+}
+
+type UnitInputs = {
+  unit_desc_id: string;
+  fatality_count: number;
+  injury_count: number;
+};
+
+type CrashInputs = {
+  case_id: string;
+  crash_timestamp: string;
+  created_by: string;
+  cris_schema_version: "2023";
+  is_temp_record: boolean;
+  private_dr_fl: boolean;
+  rpt_city_id: number;
+  rpt_sec_street_name: string;
+  rpt_street_name: string;
+  units_cris: UnitInputs[];
+  updated_by: string;
+};
+
+const DEFAULT_UNIT: UnitInputs = {
+  unit_desc_id: "",
+  fatality_count: 0,
+  injury_count: 0,
+};
+
+const DEFAULT_FORM_VALUES: CrashInputs = {
+  case_id: "",
+  crash_timestamp: "",
+  created_by: "",
+  cris_schema_version: "2023",
+  is_temp_record: true,
+  private_dr_fl: false,
+  rpt_city_id: 22,
+  rpt_sec_street_name: "",
+  rpt_street_name: "",
+  units_cris: [{ ...DEFAULT_UNIT }],
+  updated_by: "",
+};
+
+/**
+ * Construct a unit record with people records such
+ * that it can be included in a crash insert mutation
+ */
+const makeUnitRecord = (unit: UnitInputs, index: number, userEmail: string) => {
+  const unitNumber = index + 1;
+
+  const unitRecord: Record<string, unknown> = {
```

Comment:
similar to the recommendations form, I left the typings loose here because this data is heading straight to the graphql API, so I don't see a benefit in coding out the exact object shape. this seems sufficiently typesafe.